### PR TITLE
AFS remote access

### PR DESCRIPTION
**Pull request by geofjamg**

### CODE REVIEW

File: afs/afs-ws/afs-ws-client/src/main/java/com/powsybl/afs/ws/client/RemoteAppFileSystemConfig.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+            int maxAdditionalDriveCount = moduleConfig.getIntProperty("max-additional-url-count", 0);
+            for (int i = 0; i < maxAdditionalDriveCount; i++) {
+                addConfig(moduleConfig, configs, "-" + i);
+            }
+        }
+        return configs;
+    }
+
+    private static void addConfig(ModuleConfig moduleConfig, List<RemoteAppFileSystemConfig> configs, String pos) {
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-client/src/main/java/com/powsybl/afs/ws/client/RemoteAppFileSystemConfig.java]: pos should be an integer: it really strange (confusing) to prefix it with "-" in the caller method

File: afs/afs-ws/afs-ws-server-utils/pom.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-ws</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-ws-server-utils</artifactId>
+    <name>AFS WS server side utilities</name>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server-utils/pom.xml]: A description is required by maven central

File: afs/afs-ws/afs-ws-server-utils/pom.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>powsybl-afs-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server-utils/pom.xml]: Add a new line

File: afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    public ListenableAppStorage getStorage(String fileSystemName) {
+        ListenableAppStorage storage = appData.getRemotelyAccessibleStorage(fileSystemName);
+        if (storage == null) {
+            throw new WebApplicationException("App file system '" + fileSystemName + "' not found",
+                    Response.Status.NOT_FOUND);
+        }
+        return storage;
+    }
+
+    public AppFileSystem getFileSystem(String name) {
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java]: Assert that name is not null?

File: afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        Objects.requireNonNull(appData);
+        AppFileSystem fileSystem = appData.getFileSystem(name);
+        if (fileSystem == null) {
+            throw new WebApplicationException("App file system '" + name + "' not found", Response.Status.NOT_FOUND);
+        }
+        return fileSystem;
+    }
+
+    public <T extends ProjectFile> T getProjectFile(String fileSystemName, String nodeId, Class<T> clazz) {
+        Objects.requireNonNull(fileSystemName);
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java]: If you consider the previous remark, this check is not needed anymore

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-ws</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-ws-server</artifactId>
+    <name>AFS WS Server</name>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: A description is required by maven central

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-all</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.wildfly.arquillian</groupId>
+            <artifactId>wildfly-arquillian-container-embedded</artifactId>
+            <version>${wildflyarquilliancontainerembedded.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.arquillian.junit</groupId>
+            <artifactId>arquillian-junit-container</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.shrinkwrap.resolver</groupId>
+            <artifactId>shrinkwrap-resolver-impl-maven</artifactId>
+            <version>${shrinkwrap-resolver.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt</artifactId>
+            <version>${jjwt.version}</version>
+        </dependency>
+    </dependencies>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Sort dependencies by scope

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+            <groupId>${project.groupId}</groupId>
+            <artifactId>powsybl-afs-storage-api</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt</artifactId>
+            <version>${jjwt.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+    </dependencies>
+
+    <build>
+        <finalName>${project.artifactId}</finalName>
+        <plugins>
+            <!-- Arquillian -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>${maven.dependency.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+                        <phase>process-test-classes</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.wildfly</groupId>
+                                    <artifactId>wildfly-dist</artifactId>
+                                    <version>${wildfly.dist.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-server/src/main/java/com/powsybl/afs/ws/server/AppStorageApplication.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+public class AppStorageApplication extends Application {
+
+    public AppStorageApplication() {
+        BeanConfig beanConfig = new BeanConfig();
+        beanConfig.setBasePath("/rest");
+        beanConfig.setResourcePackage(AppStorageServer.class.getPackage().getName());
+        beanConfig.setScan();
+        beanConfig.setInfo(new Info()
+                .title("AFS storage API")
+                .version("1.0")
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/src/main/java/com/powsybl/afs/ws/server/AppStorageApplication.java]: This should be a constant

File: afs/afs-ws/afs-ws-server/src/main/java/com/powsybl/afs/ws/server/AppStorageServer.java [hunk 14]
```diff
@@ -0,0 +1,583 @@
...
+                    storage.addDoubleTimeSeriesData(doubleAddition.getNodeId(), doubleAddition.getVersion(),
+                                                    doubleAddition.getTimeSeriesName(), doubleAddition.getChunks());
+                    break;
+                case STRING_TIME_SERIES_CHUNKS_ADDITION:
+                    StringTimeSeriesChunksAddition stringAddition = (StringTimeSeriesChunksAddition) change;
+                    storage.addStringTimeSeriesData(stringAddition.getNodeId(), stringAddition.getVersion(),
+                                                    stringAddition.getTimeSeriesName(), stringAddition.getChunks());
+                    break;
+                default:
+                    throw new AssertionError();
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/src/main/java/com/powsybl/afs/ws/server/AppStorageServer.java]: Add a message

File: afs/afs-ws/afs-ws-storage/pom.xml [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-ws</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-ws-storage</artifactId>
+    <name>AFS WS storage</name>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-storage/pom.xml]: A description is required by maven central

File: afs/afs-ws/afs-ws-storage/pom.xml [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>powsybl-afs-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-client</artifactId>
+            <version>${resteasy.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-storage/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-storage/pom.xml [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.tyrus</groupId>
+            <artifactId>tyrus-client</artifactId>
+            <version>${tyrus.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-storage/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-storage/pom.xml [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.tyrus</groupId>
+            <artifactId>tyrus-container-jdk-client</artifactId>
+            <version>${tyrus.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-storage/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-storage/pom.xml [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>powsybl-afs-ws-utils</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-storage/pom.xml]: Dependencies are not well sorted

File: afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/RemoteAppStorage.java [hunk 20]
```diff
@@ -0,0 +1,892 @@
...
+/**
+ * @author Ali Tahanout <ali.tahanout at rte-france.com>
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class RemoteAppStorage implements AppStorage {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoteAppStorage.class);
+
+    private static final int BUFFER_MAXIMUM_CHANGE = 1000;
+    private static final long BUFFER_MAXIMUM_SIZE = Math.round(Math.pow(2, 20)); // 1Mo
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/RemoteAppStorage.java]: Seriously? Do we really need to compute the buffer size like this?

File: afs/afs-ws/afs-ws-utils/pom.xml [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-ws</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-ws-utils</artifactId>
+    <name>AFS WS Utilities</name>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-utils/pom.xml]: A description is required by maven central

File: afs/afs-ws/afs-ws-utils/pom.xml [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>javax</groupId>
+            <artifactId>javaee-api</artifactId>
+            <version>${javaeeapi.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-utils/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/afs-ws-utils/pom.xml [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.jaxrs</groupId>
+            <artifactId>jackson-jaxrs-json-provider</artifactId>
+            <version>${jackson.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-utils/pom.xml]: Do we need to specify the version here?

File: afs/afs-ws/pom.xml [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+
+    <modules>
+        <module>afs-ws-client</module>
+        <module>afs-ws-client-utils</module>
+        <module>afs-ws-server</module>
+        <module>afs-ws-server-utils</module>
+        <module>afs-ws-storage</module>
+        <module>afs-ws-utils</module>
+    </modules>
+</project>
```
Comment (mathbagu) [inline afs/afs-ws/pom.xml]: Add a new line

File: afs/afs-ws/pom.xml [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-ws</artifactId>
+    <name>AFS WS</name>
```
Comment (mathbagu) [inline afs/afs-ws/pom.xml]: A description is required by maven central

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/WriterInterceptorGzip.java [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+package com.powsybl.afs.ws.utils.gzip;
+
+import javax.ws.rs.ext.Provider;
+
+/**
+ * @author Ali Tahanout <ali.tahanout at rte-france.com>
+ */
+@Provider
+@Compress
+public class WriterInterceptorGzip extends WriterInterceptorGzipCli { }
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/WriterInterceptorGzip.java]: Is it ok with the coding style to put the braces at the end of the line like this?

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java [hunk 27]
```diff
@@ -0,0 +1,50 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReaderInterceptorGzip.class);
+
+    @Override
+    public Object aroundReadFrom(ReaderInterceptorContext interceptorContext) throws IOException {
+        String encoding = interceptorContext.getHeaders().getFirst(HttpHeaders.CONTENT_ENCODING);
+        LOGGER.trace("Encoding: {}", encoding);
+        final InputStream inputStream = interceptorContext.getInputStream();
+
+        if (encoding == null || !encoding.contains("gzip")) {
+            LOGGER.trace("Compression: NOT GZIP");
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java]: This trace is useless: the interesting information is the value of encoding, but it's already log

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java [hunk 28]
```diff
@@ -0,0 +1,50 @@
...
+            interceptorContext.setInputStream(inputStream);
+            return interceptorContext.proceed();
+        } else {
+            LOGGER.trace("Compression: read from GZIPStream");
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java]: Same remark here

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/JsonProvider.java [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * @author Ali Tahanout <ali.tahanout at rte-france.com>
+ */
+@Provider
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
+public class JsonProvider extends JacksonJaxbJsonProvider {
+
+    public JsonProvider() {
```
[the remark this answers is not in the document]
Comment (mathbagu) [reply afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/JsonProvider.java]: I think this one have to be solved or understood (JacksonJaxbJsonProvider is deprecated?)

File: afs/afs-ws/afs-ws-server-utils/pom.xml [hunk 30]
```diff
@@ -0,0 +1,40 @@
...
+
+    <artifactId>powsybl-afs-ws-server-utils</artifactId>
+    <name>AFS WS server side utilities</name>
+
+    <dependencies>
+        <!-- Compile dependencies -->
+        <dependency>
+            <groupId>javax</groupId>
+            <artifactId>javaee-api</artifactId>
+            <version>${javaeeapi.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server-utils/pom.xml]: Do we really need to specify the version here?

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 31]
```diff
@@ -0,0 +1,173 @@
...
+
+    <artifactId>powsybl-afs-ws-server</artifactId>
+    <name>AFS WS Server</name>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.jboss.arquillian</groupId>
+                <artifactId>arquillian-bom</artifactId>
+                <version>${arquillianbom.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we really need to specify the version here?

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 32]
```diff
@@ -0,0 +1,173 @@
...
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <!-- Compile dependencies -->
+        <dependency>
+            <groupId>javax</groupId>
+            <artifactId>javaee-api</artifactId>
+            <version>${javaeeapi.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we really need to specify the version here?

File: afs/afs-ws/afs-ws-server/pom.xml [hunk 33]
```diff
@@ -0,0 +1,173 @@
...
+            <groupId>org.jboss.shrinkwrap.resolver</groupId>
+            <artifactId>shrinkwrap-resolver-impl-maven</artifactId>
+            <version>${shrinkwrap-resolver.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>io.swagger</groupId>
+            <artifactId>swagger-jaxrs</artifactId>
+            <version>${swaggerjaxrs.version}</version>
```
Comment (mathbagu) [inline afs/afs-ws/afs-ws-server/pom.xml]: Do we really need to specify the version here?

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/JsonProvider.java (hunk 29 above)
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/JsonProvider.java]: I don't understand what is deprecated here, JacksonJaxbJsonProvider is not, its constructor is not too. But this is not really an issue.

File: .gitignore [hunk 34]
```diff
@@ -11,6 +11,16 @@
...
+/afs/afs-ws/afs-ws-client/target/
+/afs/afs-ws/afs-ws-client-utils/target/
+/afs/afs-ws/afs-ws-server/target/
+/afs/afs-ws/afs-ws-server-utils/target/
+/afs/afs-ws/afs-ws-storage/target/
+/afs/afs-ws/afs-ws-utils/target/
+/afs/afs-ws/target/
+/afs/afs-network/afs-network-client/target/
+/afs/afs-network/afs-network-server/target/
+/afs/afs-network/target/
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply .gitignore]: Done

File: afs/afs-network/afs-network-client/pom.xml [hunk 35]
```diff
@@ -0,0 +1,53 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-network</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-network-client</artifactId>
+    <name>Network cache AFS client</name>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-network/afs-network-client/pom.xml]: Done

File: afs/afs-network/afs-network-server/pom.xml [hunk 36]
```diff
@@ -0,0 +1,49 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-network</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-network-server</artifactId>
+    <name>Network cache AFS server</name>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-network/afs-network-server/pom.xml]: Done

File: afs/afs-network/afs-network-client/pom.xml [hunk 37]
```diff
@@ -0,0 +1,53 @@
...
+
+    <dependencies>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-network/afs-network-client/pom.xml]: ?

File: afs/afs-network/pom.xml [hunk 38]
```diff
@@ -0,0 +1,30 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-network</artifactId>
+    <name>Network cache AFS</name>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-network/pom.xml]: Done

File: afs/afs-ws/afs-ws-client-utils/pom.xml [hunk 39]
```diff
@@ -0,0 +1,47 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-ws</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-ws-client-utils</artifactId>
+    <name>AFS WS client side utilities</name>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-ws/afs-ws-client-utils/pom.xml]: Done

File: afs/afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/AbstractRemoteServiceConfig.java [hunk 40]
```diff
@@ -0,0 +1,91 @@
...
+    protected String appName;
+
+    protected int port;
+
+    protected boolean secure;
+
+    protected AbstractRemoteServiceConfig(String hostName, String appName, int port, boolean secure) {
+        this.hostName = Objects.requireNonNull(hostName);
+        this.appName = Objects.requireNonNull(appName);
+        this.port = port;
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/AbstractRemoteServiceConfig.java]: Done

File: afs/afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/AbstractRemoteServiceConfig.java [hunk 41]
```diff
@@ -0,0 +1,91 @@
...
+        try {
+            return new URI(secure ? "wss" : "ws", null, hostName, port, "/" + appName, null, null);
+        } catch (URISyntaxException e) {
+            throw new UncheckedUriSyntaxException(e);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "(hostName=" + hostName + ", appName=" + appName + ", port=" + port + ", secure=" + secure + ")";
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/AbstractRemoteServiceConfig.java]: I prefer like this

File: afs/afs-ws/afs-ws-client/pom.xml [hunk 42]
```diff
@@ -0,0 +1,60 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.powsybl</groupId>
+        <artifactId>powsybl-afs-ws</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>powsybl-afs-ws-client</artifactId>
+    <name>AFS WS client</name>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-ws/afs-ws-client/pom.xml]: Done

File: afs/afs-ws/afs-ws-client/pom.xml [hunk 43]
```diff
@@ -0,0 +1,60 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-simple</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.auto.service</groupId>
+            <artifactId>auto-service</artifactId>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply afs/afs-ws/afs-ws-client/pom.xml]: Done

File: afs/afs-ws/afs-ws-client/src/main/java/com/powsybl/afs/ws/client/RemoteAppFileSystemConfig.java (hunk 1 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-client/src/main/java/com/powsybl/afs/ws/client/RemoteAppFileSystemConfig.java]: Changed to integer

File: afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java (hunk 4 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java]: Done

File: afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java (hunk 5 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server-utils/src/main/java/com/powsybl/afs/ws/server/utils/AppDataBean.java]: Removed

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 6 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Done

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 7 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Moved to root pom dependency management

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 8 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 10 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 9 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Done

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 12 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Yes

File: afs/afs-ws/afs-ws-server/src/main/java/com/powsybl/afs/ws/server/AppStorageServer.java (hunk 14 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/src/main/java/com/powsybl/afs/ws/server/AppStorageServer.java]: Done

File: afs/afs-ws/afs-ws-storage/pom.xml (hunk 15 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-storage/pom.xml]: Done

File: afs/afs-ws/afs-ws-storage/pom.xml (hunk 16 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-storage/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-storage/pom.xml (hunk 17 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-storage/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-storage/pom.xml (hunk 18 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-storage/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-storage/pom.xml (hunk 19 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-storage/pom.xml]: Done

File: afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/RemoteAppStorage.java (hunk 20 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/RemoteAppStorage.java]: Yes

File: afs/afs-ws/afs-ws-utils/pom.xml (hunk 21 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-utils/pom.xml]: Done

File: afs/afs-ws/afs-ws-utils/pom.xml (hunk 22 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-utils/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-utils/pom.xml (hunk 23 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-utils/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/pom.xml (hunk 24 above)
Comment (geofjamg) [reply afs/afs-ws/pom.xml]: Done

File: afs/afs-ws/pom.xml (hunk 25 above)
Comment (geofjamg) [reply afs/afs-ws/pom.xml]: Done

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/WriterInterceptorGzip.java (hunk 26 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/WriterInterceptorGzip.java]: Fixed

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java (hunk 27 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java]: Removed

File: afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java (hunk 28 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-utils/src/main/java/com/powsybl/afs/ws/utils/gzip/ReaderInterceptorGzip.java]: Removed

File: afs/afs-ws/afs-ws-server-utils/pom.xml (hunk 30 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server-utils/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 31 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 32 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Added to root pom dependency management

File: afs/afs-ws/afs-ws-server/pom.xml (hunk 33 above)
Comment (geofjamg) [reply afs/afs-ws/afs-ws-server/pom.xml]: Added to root pom dependency management

**Pull request merged**